packages/nuxt/src/app/nuxt.ts

```diff
@@ -336,12 +363,62 @@ export function normalizePlugins (_plugins: Plugin[]) {
     console.warn('[warn] [nuxt] You are using a plugin that has not been wrapped in `defineNuxtPlugin`. It is advised to wrap your plugins as in the future this may enable enhancements:', unwrappedPlugins.map(p => p.name || p).join(','))
   }
 
-  return plugins as Plugin[]
+  return plugins
 }
 
-export function defineNuxtPlugin<T extends Record<string, unknown>> (plugin: Plugin<T>) {
-  plugin[NuxtPluginIndicator] = true
-  return plugin
+// -50: pre-all (nuxt)
+// -40: custom payload revivers (user)
+// -30: payload reviving (nuxt)
+// -20: pre (user) <-- pre mapped to this
+// -10: default (nuxt)
+// 0: default (user) <-- default behavior
+// +10: post (nuxt)
+// +20: post (user) <-- post mapped to this
```

Comment:
(`this` should be `pre`/`post`)

Reply:
Maybe also expose them as an (internal) map for nuxt hook numbers as constants/in a enum?

Reply:
I've deliberately not exposed them (for now) so we have freedom to change them internally.

Reply:
I'm not suggesting to expose either. Just internally make a variable for levels.

Reply:
Ah, understood. I think the only place where we use them outside this file is the payload reviver plugin. I think it's simpler not to export it as it will be exposed via `#app/nuxt` if we do.

Reply:
okay lets do it later